tools/slicec-cs/src/cs_compile.rs

```diff
@@ -62,6 +65,26 @@ fn ensure_custom_types_have_type_attribute(compilation_state: &mut CompilationSt
     }
 }
 
+fn disallow_enums_with_associated_fields(compilation_state: &mut CompilationState) {
+    for node in compilation_state.ast.as_slice() {
+        if let Node::Enum(enum_ptr) = node {
+            let enum_def = enum_ptr.borrow();
+            if enum_def.underlying.is_none() && !enum_def.supported_encodings().supports(Encoding::Slice1) {
+                let identifier = enum_def.identifier();
+                Diagnostic::new(Error::Syntax {
+                    message: "enums with associated fields are not supported by slicec-cs".to_owned(),
+                })
+                .add_note(
+                    format!("Try specifying an underlying type on your enum: `enum {identifier}: varint32`"),
```

Comment:
There is a space before the `:` with our preferred Slice formatting.

Reply:
I always forget that. Fixed.